test/framework/resources/k8s/resources/deployment.go

```diff
@@ -60,6 +60,11 @@ func (d *defaultDeploymentManager) CreateAndWaitTillDeploymentIsReady(deployment
 func (d *defaultDeploymentManager) DeleteAndWaitTillDeploymentIsDeleted(deployment *v1.Deployment) error {
 	ctx := context.Background()
 	err := d.k8sClient.Delete(ctx, deployment)
+
+	if errors.IsNotFound(err) {
```

Comment:
why this change? This looks like it will skip the normal return below

Reply:
So I was trying to make sure we end up cleaning all the resources we create during a test and not stop the cleanup just because of resource doesn't exist error. 
Say if a deployment doesn't exist ( eg: we got an error before the deployment was even created), we shouldn't error out during the cleanup process. 
Specifically, in the After each step, we check the status of delete deployment call and if it returns an not found error, the rest of the cleanup get's missed. (eg - https://github.com/aws/amazon-vpc-cni-k8s/blob/master/test/integration/cni/pod_traffic_test.go#L150)

Reply:
I see. So I think the function should still return an error, but we should do all of the cleanup regardless of whether an earlier step returns an error, similar to what we do here: https://github.com/aws/amazon-vpc-cni-k8s/blob/master/test/integration/custom-networking/custom_networking_suite_test.go#L175

Basically, we append errors and check them at the end